drivers/iio/frequency/adf41513.c

```diff
@@ -0,0 +1,1196 @@
+
+	/* Configure phase detector polarity */
+	if (st->pdata->phase_detector_polarity)
+		st->regs[ADF41513_REG6] |= ADF41513_REG6_PD_POLARITY_MSK;
+
+	/* narrow ABP | loss of lock detect enable | SD reset | LDP from pdata */
+	st->regs[ADF41513_REG6] |= ADF41513_REG6_ABP_MSK |
+				   ADF41513_REG6_LOL_ENABLE_MSK |
+				   ADF41513_REG6_SD_RESET_MSK |
+				   ADF41513_REG6_LDP(st->pdata->lock_detect_precision);
+
+	/* LD count from pdata | PS bias */
+	st->regs[ADF41513_REG7] |= ADF41513_REG7_LD_COUNT(st->pdata->lock_detect_count) |
+				   ADF41513_REG7_LD_CLK_SEL(st->pdata->fast_lock_en) |
+				   ADF41513_REG7_PS_BIAS(2);
+
+	/* enable phase resync and configure the clk divs */
+	st->regs[ADF41513_REG5] |= ADF41513_REG5_CLK1_DIV(st->pdata->phase_resync_clk_div[0]);
+	st->regs[ADF41513_REG7] |= ADF41513_REG7_CLK_DIV_MODE(st->pdata->phase_resync_en ? 2 : 0) |
+				   ADF41513_REG7_CLK2_DIV(st->pdata->phase_resync_clk_div[1]);
+
+	/* lock detect bias */
+	st->regs[ADF41513_REG9] |= ADF41513_REG9_LD_BIAS(st->pdata->lock_detect_bias);
+
+	/* power down select */
+	st->regs[ADF41513_REG11] |= ADF41513_REG11_POWER_DOWN_SEL_MSK;
+
+	/* muxout */
+	st->regs[ADF41513_REG12] |= ADF41513_REG12_MUXOUT(st->pdata->muxout_select) |
+				    ADF41513_REG12_LOGIC_LEVEL(st->pdata->muxout_1v8_en ? 0 : 1) |
+				    ADF41513_REG12_LE_SELECT(st->pdata->phase_resync_en);
+
+	/* set power-up frequency if specified */
+	if (st->pdata->power_up_frequency) {
+		ret = adf41513_set_frequency(st,
+					     (u64)st->pdata->power_up_frequency *
+						ADF41513_HZ_TO_UHZ,
+					     ADF41513_SYNC_ALL);
+		if (ret < 0) {
+			dev_err(&st->spi->dev, "Failed to set power-up frequency: %d\n", ret);
+			return ret;
+		}
+	} else {
+		/* init powered down */
+		st->regs[ADF41513_REG6] |= ADF41513_REG6_POWER_DOWN(1);
+		st->regs[ADF41513_REG12] &= ~ADF41513_REG12_LE_SELECT_MSK;
+
+		/* write initial configuration */
+		ret = adf41513_sync_config(st, ADF41513_SYNC_ALL);
+		if (ret < 0) {
+			dev_err(&st->spi->dev, "Failed to write initial config: %d\n", ret);
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+static void adf41513_power_down(void *data)
+{
+	struct adf41513_state *st = data;
+
+	guard(mutex)(&st->lock);
+	adf41513_suspend(st);
+
+	if (st->chip_enable)
+		gpiod_set_value_cansleep(st->chip_enable, 0);
+}
+
+static int adf41513_pm_suspend(struct device *dev)
+{
+	struct iio_dev *indio_dev = dev_get_drvdata(dev);
+	struct adf41513_state *st = iio_priv(indio_dev);
+
+	guard(mutex)(&st->lock);
+	return adf41513_suspend(st);
+}
+
+static int adf41513_pm_resume(struct device *dev)
+{
+	struct iio_dev *indio_dev = dev_get_drvdata(dev);
+	struct adf41513_state *st = iio_priv(indio_dev);
+
+	guard(mutex)(&st->lock);
+	return adf41513_resume(st);
+}
+
+static int adf41513_probe(struct spi_device *spi)
+{
+	struct adf41513_platform_data *pdata;
+	struct iio_dev *indio_dev;
+	struct adf41513_state *st;
+	struct clk *ref_clk;
```

Comment:
```suggestion
	struct clk *ref_clk = NULL;
```
if (pdata->clkin), ref_clk is never set.